### PR TITLE
Add RMM pool support directly from dask-cuda-worker/LocalCUDACluster

### LINKED ISSUE #233: Allow initialization of RMM upon creation of workers
To ensure a smoother operation, both from the user's perspective and internal Dask multithreading, we should make sure we can initialize RMM upon the creation of each worker.

cc @jakirkham @quasiben @kkraus14

### PR DESCRIPTION
**Pull request by pentschev**
Note for now it doesn't add support for managed memory as we can't use it with UCX at this time. If anybody thinks we absolutely need it right away, please speak up.

Fixes #233 

cc @beckernick @jakirkham

### CODE REVIEW

**Comment by beckernick** [comment]
I think hard coding managed=False is likely fine for now

**Comment by pentschev** [comment]
Codecov is complaining about the `ImportError` for `rmm` that we can't really test, so we can ignore that.

File: dask_cuda/local_cuda_cluster.py [hunk 1]
```diff
@@ -166,6 +171,16 @@ def __init__(
...
         self.device_memory_limit = device_memory_limit
 
+        self.rmm_pool_size = rmm_pool_size
+        if rmm_pool_size is not None:
+            try:
+                import rmm
+            except ImportError:
+                raise ImportError(
+                    "RMM pool requested but module 'rmm' is not available"
+                )
```
Comment (jakirkham) [inline dask_cuda/local_cuda_cluster.py]: Should make coverage happier. Not sure if we need this for all lines or just the last one.

```suggestion
                )  # pragma: no cover
```

Comment (pentschev) [reply dask_cuda/local_cuda_cluster.py]: Does that also work for codecov? I'm not sure whether it uses coverage.py or the same syntax.

Comment (jakirkham) [reply dask_cuda/local_cuda_cluster.py]: Yeah I think all Python coverage tools use `coverage.py`.

Comment (pentschev) [reply dask_cuda/local_cuda_cluster.py]: Thanks @jakirkham , it seems to work indeed!

**Comment by pentschev** [comment]
@jakirkham could you take another look at this? Should be good now.

File: dask_cuda/dask_cuda_worker.py [hunk 2]
```diff
@@ -113,6 +114,12 @@
     "'auto', or zero for no memory management "
     "(i.e., allow full device memory usage).",
 )
+@click.option(
+    "--rmm-pool-size",
+    default=None,
+    help="Initialize each worker with an RMM pool of the given size. "
+    "This can be an integer (bytes) or string (like 5GB or 5000M)."
```
Comment (jakirkham) [inline dask_cuda/dask_cuda_worker.py]: Should we say something about it being disabled by default?

File: dask_cuda/local_cuda_cluster.py [hunk 3]
```diff
@@ -166,6 +171,16 @@ def __init__(
         self.host_memory_limit = memory_limit
         self.device_memory_limit = device_memory_limit
 
+        self.rmm_pool_size = rmm_pool_size
+        if rmm_pool_size is not None:
+            try:
+                import rmm
+            except ImportError:
+                raise ValueError(
+                    "RMM pool requested but module 'rmm' is not available"
```
Comment (jakirkham) [inline dask_cuda/local_cuda_cluster.py]: Should we mention how to install it?

File: dask_cuda/utils.py [hunk 4]
```diff
@@ -16,6 +16,19 @@ def setup(self, worker=None):
...
+class RMMPool:
+    def __init__(self, nbytes):
+        self.nbytes = nbytes
+
+    def setup(self, worker=None):
+        if self.nbytes is not None:
+            import rmm
+
+            rmm.reinitialize(
+                pool_allocator=True, managed_memory=False, initial_pool_size=self.nbytes
```
Comment (jakirkham) [inline dask_cuda/utils.py]: I think you already said this. So apologies if I'm rehashing things. dask-cuda sets up device `0` to be the device used by the worker right? By default this is what `rmm` will use. If not, we might need to specify it.

File: dask_cuda/dask_cuda_worker.py (hunk 2 above)
Comment (pentschev) [reply dask_cuda/dask_cuda_worker.py]: Done.

File: dask_cuda/local_cuda_cluster.py (hunk 3 above)
Comment (pentschev) [reply dask_cuda/local_cuda_cluster.py]: Done.

File: dask_cuda/utils.py (hunk 4 above)
Comment (pentschev) [reply dask_cuda/utils.py]: That's right, the relevant GPU will always be GPU 0 for every dask-cuda worker, so we don't need to specify it.

**Comment by pentschev** [comment]
Thanks @jakirkham and @madsbk for reviews.

**Comment by jakirkham** [comment]
Thanks for adding this feature Peter! 😄

**Pull request merged**